frontend/src/model/beta/releases/DraftNewReleaseDialog.tsx

```diff
@@ -43,34 +44,49 @@ export default function DraftNewReleaseDialog({
   const [isMinorRelease, setIsMinorRelease] = useState(false)
   const [artefacts, setArtefacts] = useState<File[]>([])
   const [errorMessage, setErrorMessage] = useState('')
+  const [loading, setLoading] = useState(false)
 
   async function onSubmit(event: FormEvent<HTMLFormElement>) {
     event.preventDefault()
     setErrorMessage('')
+    setLoading(true)
     if (!model.card.version) {
+      setLoading(false)
       return setErrorMessage('Please make sure your model has a schema set before drafting a release.')
     }
     if (isValidSemver(semanticVersion)) {
+      const fileIds: string[] = []
+      for (const artefact of artefacts) {
+        const postArtefactResponse = await postArtefact(artefact, model.id, artefact.name, artefact.type)
+        if (postArtefactResponse.status === 200) {
```

Comment:
You don't provide any warnings if an error does occur.  You also shouldn't check status is equal to 200, there's the `ok` attribute to determine if a request passed.